app/assets/javascripts/components/SearchOptions.tsx

```diff
@@ -0,0 +1,155 @@
+import { AppState } from '@/ui_models/app_state';
+import { toDirective, useAutorunValue } from './utils';
+import { useRef, useState } from 'preact/hooks';
+import { WebApplication } from '@/ui_models/application';
+import { ComponentChildren } from 'preact';
+import {
+  CustomCheckboxContainer,
+  CustomCheckboxInput,
+  CustomCheckboxInputProps,
+} from '@reach/checkbox';
+import '@reach/checkbox/styles.css';
+import VisuallyHidden from '@reach/visually-hidden';
+import TuneIcon from '../../icons/ic_tune.svg';
+import {
+  Disclosure,
+  DisclosureButton,
+  DisclosurePanel,
+} from '@reach/disclosure';
+import React, { HTMLProps } from 'react';
+
+type Props = {
+  appState: AppState;
+  application: WebApplication;
+};
+
+function Switch(
```

Comment:
Should be extracted into its own file no? Will need to be done when we need switches elsewhere anyway.

Reply:
True. This was originally supposed to live in StyleKit but the React components export is weirdly broken so I postponed the extraction…